drfpasswordless/serializers.py

```diff
@@ -218,40 +218,40 @@ def validate(self, attrs, user=None):
+            tokens = CallbackToken.objects.filter(**{'user': user,
                                                  'key': callback_token,
                                                  'type': CallbackToken.TOKEN_TYPE_AUTH,
                                                  'to_alias_type': alias_type.upper(),
-                                                 'to_alias': attrs.get('mobile', None) or attrs.get('email', None),
-                                                 'is_active': True})
+                                                 'to_alias': attrs.get('mobile', None) or attrs.get('email', None),})
 
-            if token.user == user:
-                # Check the token type for our uni-auth method.
-                # authenticates and checks the expiry of the callback token.
-                if not user.is_active:
-                    msg = _('User account is disabled.')
-                    raise serializers.ValidationError(msg)
+            if not tokens.count():
+                msg = _(str(InvalidCallbackToken()))
+                raise serializers.ValidationError(msg)
 
-                if api_settings.PASSWORDLESS_USER_MARK_EMAIL_VERIFIED \
-                        or api_settings.PASSWORDLESS_USER_MARK_MOBILE_VERIFIED:
-                    # Mark this alias as verified
-                    user = User.objects.get(pk=token.user.pk)
-                    success = verify_user_alias(user, token)
+            token = tokens.get(**{'is_active': True})
 
-                    if success is False:
-                        msg = _('Error validating user alias.')
-                        raise serializers.ValidationError(msg)
+            # Check the token type for our uni-auth method.
+            # authenticates and checks the expiry of the callback token.
+            if not user.is_active:
+                msg = _('User account is disabled.')
+                raise serializers.ValidationError(msg)
 
-                attrs['user'] = user
-                token.delete()
-                return attrs
+            if api_settings.PASSWORDLESS_USER_MARK_EMAIL_VERIFIED \
+                    or api_settings.PASSWORDLESS_USER_MARK_MOBILE_VERIFIED:
+                # Mark this alias as verified
+                user = User.objects.get(pk=token.user.pk)
+                success = verify_user_alias(user, token)
+
+                if success is False:
+                    msg = _('Error validating user alias.')
+                    raise serializers.ValidationError(msg)
+
+            attrs['user'] = user
+            token.delete()
+            return attrs
 
-            else:
-                msg = _('Invalid Token')
-                raise serializers.ValidationError(msg)
         except CallbackToken.DoesNotExist:
-            msg = _(str(InvalidCallbackToken()))
-            raise serializers.ValidationError(msg)
+            raise AuthenticationFailed()
         except User.DoesNotExist:
```

Comment:
if you change 

```python
user = User.objects.get(pk=token.user.pk)
```

then make sure to change catching to the KeyError and check what key was used (or raise above instead)